### PR TITLE
refactor(git): use openssl for git

### LINKED ISSUE #1250: GnuTLS recv error in git resolver
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

When clone code, this happened:

```
error: RPC failed; curl 56 GnuTLS recv error (-9): Error decoding the received TLS packet.
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

[Rebuilding git with openssl instead of gnutls](https://stackoverflow.com/questions/38378914/git-error-rpc-failed-curl-56-gnutls)

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #1250

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: build/resolver/git/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,12 @@
-FROM ubuntu:18.10
+FROM alpine:3.8
```
Comment (supereagle) [inline build/resolver/git/Dockerfile]: Unify base image for all resolvers?

File: build/resolver/git/Dockerfile [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
+LABEL maintainer="zhujian@caicloud.io"
 
 ENV WORKDIR /workspace
 WORKDIR $WORKDIR
 
-RUN apt-get update \
-    && apt-get install -y git \
-    && rm -rf /var/lib/apt/lists/*
+RUN apk update && apk add ca-certificates && \
+    apk add --no-cache git bash coreutils
```
[the remark this answers is not in the document]
Comment (zhujian7) [reply build/resolver/git/Dockerfile]: fixed

File: build/resolver/git/Dockerfile (hunk 1 above)
Comment (zhujian7) [reply build/resolver/git/Dockerfile]: Will fix by another PR.

**Pull request merged**